velox/functions/prestosql/tests/GreatestLeastTest.cpp

```diff
@@ -83,29 +92,132 @@ TEST_F(GreatestLeastTest, leastReal) {
       {0, -100, -1.1});
 }
 
-TEST_F(GreatestLeastTest, nanInput) {
-  // Presto rejects NaN inputs of type DOUBLE, but allows NaN inputs of type
-  // REAL.
-  std::vector<double> input{0, 1.1, std::nan("1")};
-  VELOX_ASSERT_THROW(
-      runTest<double>("least(c0)", {{0.0 / 0.0}}, {0}),
-      "Invalid argument to least(): NaN");
-  runTest<double>("try(least(c0, 1.0))", {input}, {0, 1.0, std::nullopt});
-
-  VELOX_ASSERT_THROW(
-      runTest<double>("greatest(c0)", {1, {0.0 / 0.0}}, {1, 0}),
-      "Invalid argument to greatest(): NaN");
-  runTest<double>("try(greatest(c0, 1.0))", {input}, {1.0, 1.1, std::nullopt});
-
-  auto result = evaluateOnce<bool, float, float>(
-      "is_nan(least(c0))", std::nanf("1"), 1.2);
-  ASSERT_TRUE(result.has_value());
-  ASSERT_TRUE(result.value());
-
-  result = evaluateOnce<bool, float, float>(
-      "is_nan(greatest(c0))", std::nanf("1"), 1.2);
-  ASSERT_TRUE(result.has_value());
-  ASSERT_TRUE(result.value());
+TEST_F(GreatestLeastTest, leastNanInput) {
+  runTest<float>(
```

Comment:
These tests are a bit verbose and difficult to read.

Consider refactoring:

```
  auto greatest1 = [&](float a) {
    return evaluateOnce<float, float>("greatest(c0)", {a}).value();
  };

  auto greatest2 = [&](float a, float b) {
    return evaluateOnce<float, float, float>("greatest(c0, c1)", {a}, {b})
        .value();
  };

  auto least1 = [&](float a) {
    return evaluateOnce<float, float>("least(c0)", {a}).value();
  };

  auto least2 = [&](float a, float b) {
    return evaluateOnce<float, float, float>("least(c0, c1)", {a}, {b}).value();
  };

  EXPECT_EQ(greatest1(1.0), 1.0);
  EXPECT_EQ(greatest2(1.0, 2.0), 2.0);
  EXPECT_EQ(least1(1.0), 1.0);
  EXPECT_EQ(least2(1.0, 2.0), 1.0);
```

Reply:
Fixed